source/replication/enable-server-side-two-way-bucket-replication.rst

```diff
@@ -0,0 +1,514 @@
+      for a remote target to synchronize data into *any* bucket in the MinIO
+      cluster. To restrict the policy to specific buckets, specify those 
+      buckets as an element in the ``Resource`` array similar to 
+      ``"arn:aws:s3:::bucketName/*"``.
+
+      Use the :mc-cmd:`mc admin policy set` command to associate the policy 
+      to a user on the target MinIO cluster.
+
+MinIO strongly recommends creating users specifically for supporting 
+bucket replication operations. See 
+:mc:`mc admin user` and :mc:`mc admin policy` for more complete
+documentation on adding users and policies to a MinIO cluster.
+
+Considerations
+--------------
+
+Replication of Existing Objects
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+MinIO performs replication as part of writing an object (PUT operations). MinIO
+does *not* apply replication rules to existing objects in the bucket. Use
+:mc:`mc cp` or :mc:`mc mirror` to migrate existing objects to the destination
+bucket.
+
+For buckets with active write operations during the procedure, any objects
+written *before* configuring bucket replication remain unreplicated. 
+
+Consider scheduling a maintenance period during which applications stop
+all write operations to the bucket or buckets for which you are configuring
+bucket replication. Restart write operations at the completion of the
+procedure to ensure consistent object replication.
+
+Replication of Delete Operations
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+MinIO supports replicating delete operations onto the target bucket. 
+Specifically, MinIO can replicate versioning
+:s3-docs:`Delete Markers <versioning-workflows.html>` and the deletion
+of specific versioned objects:
+
+- For delete operations on an object, MinIO replication also creates the delete
+  marker on the target bucket.
+
+- For delete operations on versions of an object,
+  MinIO replication also deletes those versions on the target bucket.
+
+MinIO requires explicitly enabling replication of delete operations using the
+:mc-cmd-option:`mc replicate add replicate` or 
+:mc-cmd-option:`mc replicate edit replicate`. This procedure includes the
+required flags for enabling replication of delete operations and delete markers.
+
+Replication of Encrypted Objects
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+MinIO supports replicating objects encrypted with automatic 
+Server-Side Encryption (SSE-S3). Both the source and destination buckets
+*must* have automatic SSE-S3 enabled for MinIO to replicate an encrypted object.
+
+As part of the replication process, MinIO *decrypts* the object on the source
+bucket and transmits the unencrypted object. The destination MinIO cluster then
+re-encrypts the object using the destination bucket SSE-S3 configuration. MinIO
+*strongly recommends* :ref:`enabling TLS <minio-TLS>` on both source and
+destination clusters to ensure the safety of objects during transmission.
+
+MinIO does *not* support replicating client-side encrypted objects 
+(SSE-C).
+
```

Comment:
Might want to mention that replication between object lock enabled buckets is possible only when both sides have object-locking enabled. Also needs additional object-locking, legal hold/retention permissions in the policy statement

Reply:
Hm. This would also be true for encryption. Do we have a list anywhere of the required permissions? I think it would require added perms for both the source replication admin and the destination replication user

Reply:
Done!